pkg/discovery/oci_dbbacked_test.go

```diff
@@ -0,0 +1,411 @@
+					Arch:   "amd64",
+				},
+				{
+					Image:  "darwin/arm64/k8s/cluster:v0.28.0",
+					URI:    "",
+					Digest: "1b44444",
+					OS:     "darwin",
+					Arch:   "arm64",
+				},
+			},
+		},
+	},
+	{
+		Name:               "cluster",
+		Target:             configtypes.TargetTMC,
+		Description:        "Cluster plugin for tmc",
+		Publisher:          "tmc",
+		Vendor:             "vmware",
+		RecommendedVersion: "v0.2.0",
+		Artifacts: distribution.Artifacts{
+			"v0.0.1": []distribution.Artifact{
+				{
+					Image:  "darwin/amd64/k8s/cluster:v0.0.1",
+					URI:    "",
+					Digest: "2a11111",
+					OS:     "darwin",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "linux/amd64/k8s/cluster:v0.0.1",
+					URI:    "",
+					Digest: "2a22222",
+					OS:     "linux",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "windows/amd64/k8s/cluster:v0.0.1",
+					URI:    "",
+					Digest: "2a33333",
+					OS:     "windows",
+					Arch:   "amd64",
+				},
+			},
+			"v0.2.0": []distribution.Artifact{
+				{
+					Image:  "darwin/amd64/k8s/cluster:v0.2.0",
+					URI:    "",
+					Digest: "2b11111",
+					OS:     "darwin",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "linux/amd64/k8s/cluster:v0.2.0",
+					URI:    "",
+					Digest: "2b22222",
+					OS:     "linux",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "windows/amd64/k8s/cluster:v0.2.0",
+					URI:    "",
+					Digest: "2b33333",
+					OS:     "windows",
+					Arch:   "amd64",
+				},
+			},
+		},
+	},
+	{
+		Name:               "telemetry",
+		Target:             configtypes.TargetK8s,
+		Description:        "telemetry plugin for k8s",
+		Publisher:          "tkg",
+		Vendor:             "vmware",
+		RecommendedVersion: "v0.28.0",
+		Artifacts: distribution.Artifacts{
+			"v0.26.0": []distribution.Artifact{
+				{
+					Image:  "darwin/amd64/k8s/telemetry:v0.26.0",
+					URI:    "",
+					Digest: "3a11111",
+					OS:     "darwin",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "linux/amd64/k8s/telemetry:v0.26.0",
+					URI:    "",
+					Digest: "3a22222",
+					OS:     "linux",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "windows/amd64/k8s/telemetry:v0.26.0",
+					URI:    "",
+					Digest: "3a33333",
+					OS:     "windows",
+					Arch:   "amd64",
+				},
+			},
+			"v0.28.0": []distribution.Artifact{
+				{
+					Image:  "darwin/amd64/k8s/telemetry:v0.28.0",
+					URI:    "",
+					Digest: "3b11111",
+					OS:     "darwin",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "linux/amd64/k8s/telemetry:v0.28.0",
+					URI:    "",
+					Digest: "3b22222",
+					OS:     "linux",
+					Arch:   "amd64",
+				},
+				{
+					Image:  "windows/amd64/k8s/telemetry:v0.28.0",
+					URI:    "",
+					Digest: "3b33333",
+					OS:     "windows",
+					Arch:   "amd64",
+				},
+			},
+		},
+	},
+}
+
+type stubInventory struct{}
+
+func (stub *stubInventory) GetAllPlugins() ([]*plugininventory.PluginInventoryEntry, error) {
+	return pluginEntries, nil
+}
+
+// nolint: gocyclo
```

Comment:
This stub filtering function is too complex for the linter 😄 
But I had spent enough time getting those tests to run, so I just disabled the linter for it.

This is why it is better to use SQL than our own Go logic for this smart filtering 😉